src/moin/storage/middleware/protecting.py

```diff
@@ -73,6 +85,10 @@ def __init__(self, indexer, user, acl_mapping):
         self.eval_acl = lru_cache_decorator(self._eval_acl)
         lru_cache_decorator = lru_cache(LOOKUP_CACHE)
         self.get_acls = lru_cache_decorator(self._get_acls)
+        lru_cache_decorator = lru_cache(ACL_CACHE)
+        self.allows = lru_cache_decorator(self._allows)
+        # placeholder to show we are passing meta data around without effecting lru caches
```

Comment:
typo "affecting"?

Reply:
also not sure i understand that...

it seems you have remove usage of rev.meta and rather use rev as a dict-like object.

guess that is somehow departing from the initial idea of middlewares somehow exposing the same or at least similar interface, so that one can use the middlewares or not.

but, besides that, if the idea is not to use .meta any more, shouldn't it rather be gone and raise an AttributeError?

Reply:
I have given up on using windex instead of meta, seems meta is used everywhere as the name for whoosh index data.

Will fix the use of rev/revs vs. meta/metas in frontend/views.py and admin/views.py in next commit. So revs will be used for Items, metas for whoosh indexes (dict).

Several reports (History, Index, Tags, MyChanges...) require only whoosh index data. These run much slower (10-20 times slower, see #946) if Item instances (with an open data file) are returned. If that is exposing a dissimilar interface, it is necessary.

Will fix affecting.